### PR TITLE
block: qcow: support compressed image

### PR DESCRIPTION
**Pull request by dogonthemoon**
This PR does the following:
1 Introduce ClusterType, divide cluster into five type, cluster type decides it's read/write logic。
2 Support compressed image(zlib based compression). 
3 Chose libdeflater as the decompression library, it performs best.
4 Add simple test of cluster.

### CODE REVIEW

**Comment by dogonthemoon** [comment]
@rbradford I'm sorry, i don't know how to add reviewers for this pull request, would you like to review it?

File: block/Cargo.toml [hunk 1]
```diff
@@ -25,3 +25,4 @@ virtio-queue = "0.10.0"
 vm-memory = { version = "0.13.1", features = ["backend-mmap", "backend-atomic", "backend-bitmap"] }
 vm-virtio = { path = "../vm-virtio" }
 vmm-sys-util = "0.11.0"
+libdeflater = "1.19.0"
```
Comment (rbradford) [inline block/Cargo.toml:28]: The list of crates should be in lexical order.

File: block/src/qcow/mod.rs [hunk 2]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE-BSD-3-Clause file.
 
+extern crate libdeflater;
```
Comment (rbradford) [inline block/src/qcow/mod.rs:5]: This is not required with our version of Rust.

**Review by rbradford — CHANGES_REQUESTED** [review]
I'm not sure if we really want this functionality - we've been discussing removing QCOW2 support in general as it is complex code that nobody on the team has an interesting in maintaining.

Also I wouldn't want use to use the libdeflater crate and prefer a native Rust implementation over one that incorporates a library dependency.

**Comment by dogonthemoon** [comment]
> I'm not sure if we really want this functionality - we've been discussing removing QCOW2 support in general as it is complex code that nobody on the team has an interesting in maintaining.
> 
> Also I wouldn't want use to use the libdeflater crate and prefer a native Rust implementation over one that incorporates a library dependency.

Yes I agree, QCOW2 has complex code, but it has some very import features that makes it stay alive, including this compression feature.

A native Rust implementation about decompression, would you like to give me more suggestion? 

thank you very much!

**Pull request closed without merging**